OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1413 @@
+        "bindIsolatedService", "bindServiceAsUser", "startForegroundService")
+    val broadcastReceiverStartMethods: List[String] = List("sendBroadcast", "sendOrderedBroadcast", "sendStickyBroadcast",
+        "sendStickyOrderedBroadcast", "getBroadcast", "sendBroadcastAsUser", "sendStickyBroadcastAsUser",
+        "sendStickyOrderedBroadcastAsUser")
+
+    final val onStart = "onStart"
+    final val onPause = "onPause"
+    final val onStop = "onStop"
+    final val onDestroy = "onDestroy"
+    final val onActivityResult = "onActivityResult"
+    final val onResume = "onResume"
+    final val onRestart = "onRestart"
+    final val onStartCommand = "onStartCommand"
+    final val startService = "startService"
+    final val onBind = "onBind"
+    final val onRebind = "onRebind"
+    final val onUnbind = "onUnbind"
+    final val bindService = "bindService"
+    final val stopService = "stopService"
+    final val unbindService = "unbindService"
+    final val stopSelf = "stopSelf"
+    final val onReceive = "onReceive"
+    final val onCreate = "onCreate"
+    final val onBroadcastReceive = "onBroadcastReceive"
+
+
+    //explicit
+    final val setClass = "setClass"
+    final val setClassName = "setClassName"
+    final val setComponent = "setComponent"
+
+    final val intentMethods = List(setAction, addCategory, setData, setDataAndNormalize, setDataAndType,
+        setDataAndTypeAndNormalize, setType, setTypeAndNormalize, setPackage)
+    final val explicitMethods = List(setClass, setClassName, setComponent)
+
+    val intentFilters: ListBuffer[intentFilter] = ListBuffer.empty[intentFilter]
+    val edges: ListBuffer[((Method, Int), Method)] = ListBuffer.empty[((Method, Int), Method)]
+    val explicitIntents: ListBuffer[explicitIntent] = ListBuffer.empty[explicitIntent]
+    val implicitIntents: ListBuffer[implicitIntent] = ListBuffer.empty[implicitIntent]
+
+    val senderMap: mutable.Map[(ObjectType, String), ListBuffer[(Method, String, Int)]] =
+        mutable.Map.empty[(ObjectType, String), ListBuffer[(Method, String, Int)]] //Class and name of the Method that returns the searched intent -> the intent sending methods and the component type
+    val returningIntentsMap: mutable.Map[(ObjectType, String), ListBuffer[implicitIntent]] =
+        mutable.Map.empty[(ObjectType, String), ListBuffer[implicitIntent]] //class and method that returns an intent -> intent
+
+    val componentMap: mutable.Map[String, ListBuffer[ClassFile]] = mutable.Map(
+        activity -> ListBuffer.empty[ClassFile],
+        service -> ListBuffer.empty[ClassFile]
+    )
+
+    /**
+     * Calls all necessary methods to generate the call graph edges that are Android specific.
+     *
+     * @param manifest AndroidManifest.xml
+     *
+     */
+    def performAnalysis(
+                         manifest: Elem
+    ): ProperPropertyComputationResult = {
+        parseManifest(manifest)
```

Comment:
I would strongly suggest a more functional programming style here instead of imperatively invoking several methods that manipulate some state that probably isn't required elsewhere.